app/src/forms/common/scheduleService.js

```diff
@@ -0,0 +1,597 @@
+  if (isDateInRange(today, openDate, endDate, timezone, ignoreTime)) {
+    // Check if in late submission period
+    const isLate = allowLateSubmissions && closeDate && !isDateInRange(today, openDate, closeDate, timezone, ignoreTime);
+
+    return {
+      state: 1,
+      index: 0,
+      dates: dates[0],
+      old_dates: null,
+      late: isLate ? 1 : 0,
+    };
+  }
+
+  // Not in period, determine if before or after
+  const isBeforePeriod = isDateInFuture(openDate, timezone, today, ignoreTime);
+
+  return {
+    state: isBeforePeriod ? -1 : 0,
+    index: -1,
+    dates: dates[0], // Return the period as a reference
+    old_dates: null,
+    late: -1,
+  };
+}
+
+/**
+ * Determines what type of reminder email should be sent based on current date and schedule
+ * @param {Object} scheduleOrReport The form schedule or period report
+ * @param {Date|String} referenceDate Optional reference date (defaults to now)
+ * @param {Boolean} respectTimeComponent Whether to respect time components in dates (defaults to false)
+ * @returns {String|undefined} Email type constant or undefined if no reminder should be sent
+ */
+function getEmailReminderType(scheduleOrReport, referenceDate = null, respectTimeComponent = false) {
+  // Get the current period if a schedule was provided
+  const report = scheduleOrReport.scheduleType !== undefined ? getCurrentPeriod(scheduleOrReport, referenceDate, respectTimeComponent) : scheduleOrReport;
+
+  if (!report || !report.dates) return undefined;
+
+  const now = referenceDate || new Date();
+  const timezone = report.dates.timezone || 'America/Vancouver';
+  const compareTime = respectTimeComponent;
+
+  // Form opens today - exact date match required
+  if (isSameDay(now, report.dates.startDate, timezone, compareTime)) {
+    return EmailTypes.REMINDER_FORM_OPEN;
+  }
+
+  // No close date or period is too short for mid-reminders
+  if (!report.dates.closeDate) return undefined;
+
+  const daysBetweenDates = daysBetween(report.dates.startDate, report.dates.closeDate, timezone, !respectTimeComponent);
+
+  if (daysBetweenDates <= 3) return undefined;
+
+  // Form closes tomorrow - exact date match required
+  const dayBeforeClose = calculateDatePlus(report.dates.closeDate, -1, 'days', timezone, null, !respectTimeComponent);
+  if (isSameDay(now, dayBeforeClose, timezone, compareTime)) {
+    return EmailTypes.REMINDER_FORM_WILL_CLOSE;
+  }
+
+  // Middle of the period reminder - exact date match required
+  const middleDate = calculateMiddleDate(report.dates.startDate, report.dates.closeDate, timezone, null, !respectTimeComponent);
+  if (middleDate && isSameDay(now, middleDate, timezone, compareTime)) {
+    return EmailTypes.REMINDER_FORM_NOT_FILL;
+  }
+
+  return undefined;
+}
+
+/**
+ * Validate a form schedule object
+ * @param {Object} schedule The schedule object to validate
+ * @returns {Object} Validation result {message, status}
+ */
+function validateScheduleObject(schedule = {}) {
```

Comment:
validateScheduleObject, isLateSubmissionConfigValid and isClosingMessageValid don't belong in this file. They can use functions from this file (formatting, date calculations etc). I think these belong in form.service - whereever we are validating the form data.